### PR TITLE
[skip-ci] Add docs for SSL state methods

### PR DESCRIPTION
**Pull request by jon-oracle**
##### Checklist
- [x] documentation is added or updated

##### Description of change
Added documentation for some SSL APIs

### CODE REVIEW

File: doc/man3/SSL_get_state.pod [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+ int SSL_is_init_finished(SSL *s);
+
+=head1 DESCRIPTION
+
+SSL_get_state() gets the current B<ssl> handshake state.
+
+SSL_in_before() checks if an B<ssl> handshake has started.
+
+SSL_in_init() checks if an B<ssl> handshake is in progress.
```
Comment (levitte) [inline doc/man3/SSL_get_state.pod]: I think that to the new reader, it's not clear what the actual difference is between `SSL_in_before()` and `SSL_in_init()`...  'cause when the handshake has started, the handshake is logically in progress, isn't it?

Comment (levitte) [inline doc/man3/SSL_get_state.pod]: If I understand correctly, `SSL_in_before()` is kind of like `SSL_in_init()` but indicates that no wire communication has been made yet.

**Comment by mattcaswell** [comment]
Darn. Duplicated work. See #3051

**Comment by mattcaswell** [comment]
Merged #3051 instead of this one. Thanks for the contribution anyway!

**Pull request closed without merging**